provider/provider_test.go

```diff
@@ -87,3 +125,678 @@ func TestProviderCdpRegion(t *testing.T) {
 		t.Fatalf("Terraform provider should have set cdp region. Got: %v", cdpRegion)
 	}
 }
+
+func TestGetCdpConfigCdpAccessKeyId(t *testing.T) {
+	tests := []cdpConfigTestScenarioForString{
+		{
+			base: cdpConfigTestScenarioBase{
+				name:     "test when CdpAccessKeyId is not empty",
+				envVar:   "CDP_ACCESS_KEY_ID",
+				envValue: "",
+			},
+			modelValue:     types.StringValue("some CdpAccessKeyId value"),
+			expectedResult: "some CdpAccessKeyId value",
+		},
+		{
+			base: cdpConfigTestScenarioBase{
+				name:     "test when CdpAccessKeyId is empty",
+				envVar:   "CDP_ACCESS_KEY_ID",
+				envValue: "some CdpAccessKeyId env variable value",
+			},
+			modelValue:     types.StringNull(),
+			expectedResult: "some CdpAccessKeyId env variable value",
+		},
+		{
+			base: cdpConfigTestScenarioBase{
+				name:     "test when both CdpAccessKeyId and CDP_ACCESS_KEY_ID contains no value",
+				envVar:   "CDP_ACCESS_KEY_ID",
+				envValue: "",
+			},
+			modelValue:     types.StringNull(),
+			expectedResult: "",
+		},
+	}
+	for _, test := range tests {
+		t.Setenv(test.base.envVar, test.base.envValue)
+		t.Run(test.base.name, func(t *testing.T) {
+			model := createCdpProviderModel()
+			model.CdpAccessKeyId = test.modelValue
+
+			config := getCdpConfig(context.Background(), model, testVersion, testTerraformVersion)
+
+			cred := config.Credentials
+			if cred.AccessKeyId != test.expectedResult {
+				t.Errorf("AccessKeyId diverges from the expected! Expected: %s, got: %s\n",
+					test.expectedResult, cred.AccessKeyId)
+			}
+		})
+		t.Setenv(test.base.envVar, "")
```

Comment:
You probably want to move this to a defer block and immediately after the Setenv. Also we should use unset.